app/scripts/views/mixins/password-strength-mixin.js

```diff
@@ -0,0 +1,53 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+define([
+  'lib/promise',
+  'lib/runtime-require'
+], function (p, runtimeRequire) {
+  'use strict';
+
+  var PasswordStrengthMixin = {
+    _getPasswordStrengthChecker: function () {
+      // returns a promise that resolves once the library is loaded.
+      var self = this;
+      if (! self._passwordStrengthCheckerPromise) {
+        self._passwordStrengthCheckerPromise = runtimeRequire('passwordcheck')
+          .then(function (PasswordCheck) {
+            return new PasswordCheck();
+          });
+      }
+      return self._passwordStrengthCheckerPromise;
+    },
+    /**
+     * Check the password strength. Returns a promise that resolves
+     * when the check is complete.
+     *
+     * Usage:
+     *
+     *   view.checkPasswordStrength(password)
+     *     .then(function (status) {
+     *      // do something with the status
+     *     });
+     *
+     * @method checkPasswordStrength
+     * @returns {Promise}
+     */
+    checkPasswordStrength: function (password) {
+      var self = this;
+      return this._getPasswordStrengthChecker()
+        .then(function (passwordStrengthChecker) {
+          var deferred = p.defer();
```

Comment:
I'm a bit confused about `NOT_STRONG_ENOUGH`, if my password is "not strong enough password" is that a weak password?

If that is a weak password, does this make sense for experiment results:

Z = (experiment.pw-strength.NOT_STRONG_ENOUGH / experiment.pw-strength.ENGAGED) * 100
IF Z is low - means we should not invest into UX to improve password strength 
If Z is high - we tell users their password is weak and suggest users to include a number or a letter in their password

Y = (experiment.pw-strength.BLOOMFILTER_HIT / experiment.pw-strength.engaged) * 100
IF Y is low - bloomfilter is not useful, remove it
IF Y is high - bloomfilter is useful, we should make it better and use it for UX  
(Another graph of this nature can use `BLOOMFILTER_MISS` follows the same logic)

I don't see any value in `PASSWORD_NOT_A_STRING`, `MISSING_PASSWORD` and `PASSWORD_TOO_SHORT`, any ideas?